src/test-run/bookmark.js

```diff
@@ -0,0 +1,97 @@
+import ClientFunctionBuilder from '../client-functions/client-function-builder';
+import TEST_RUN_PHASE from '../test-run/phase';
+
+import {
+    SwitchToMainWindowCommand,
+    SwitchToIframeCommand,
+    SetNativeDialogHandlerCommand,
+    SetTestSpeedCommand,
+    NavigateToCommand
+} from './commands/actions';
+
+class TestRunBookmark {
+    constructor (testRun) {
+        this.testRun = testRun;
+
+        this.url            = 'about:blank';
+        this.dialogHandler  = testRun.activeDialogHandler;
+        this.iframeSelector = testRun.activeIframeSelector;
+        this.speed          = testRun.speed;
+        this.ctx            = testRun.ctx;
+        this.fixtureCtx     = testRun.fixtureCtx;
+    }
+
+    async _restoreDialogHandler () {
+        if (this.testRun.activeDialogHandler !== this.dialogHandler) {
+            var restoreDialogCommand = new SetNativeDialogHandlerCommand({ dialogHandler: { fn: this.dialogHandler } });
+
+            await this.testRun.executeCommand(restoreDialogCommand);
+        }
+    }
+
+    async _restoreSpeed () {
+        if (this.testRun.speed !== this.speed) {
+            var restoreSpeedCommand = new SetTestSpeedCommand({ speed: this.speed });
+
+            await this.testRun.executeCommand(restoreSpeedCommand);
+        }
+    }
+
+    async _restoreWorkingFrame () {
+        if (this.testRun.activeIframeSelector !== this.iframeSelector) {
+            var switchWorkingFrameCommand = this.iframeSelector ?
+                                            new SwitchToIframeCommand({ selector: this.iframeSelector }) :
+                                            new SwitchToMainWindowCommand();
+
+            await this.testRun.executeCommand(switchWorkingFrameCommand);
+        }
+    }
+
+    async _restorePage () {
+        var navigateCommand = new NavigateToCommand({ url: this.url });
+
+        await this.testRun.executeCommand(navigateCommand);
+    }
+
+    async restore (callsite) {
+        var prevPhase = this.testRun.phase;
+
+        this.testRun.phase = TEST_RUN_PHASE.inBookmarkRestore;
```

Comment:
It's not. How do you imagine that? One-line function that does exactly the same? And it's fine to modify test run phase outside of test run - it's a public property. Moreover, it's already done in `fixture-hook-tracker`.